addons/hr_attendance/hr_attendance.py

```diff
@@ -48,7 +48,8 @@ class hr_attendance(osv.osv):
     def _day_compute(self, cr, uid, ids, fieldnames, args, context=None):
         res = dict.fromkeys(ids, '')
         for obj in self.browse(cr, uid, ids, context=context):
-            res[obj.id] = time.strftime('%Y-%m-%d', time.strptime(obj.name, '%Y-%m-%d %H:%M:%S'))
+            date_start = fields.datetime.context_timestamp(cr, uid, datetime.strptime(obj.name, '%Y-%m-%d %H:%M:%S'), context=context)
+            res[obj.id] = date_start.strftime('%Y-%m-%d')
```

Comment:
May be you can use context_today only, something like res[obj.id] = fields.date.context_today(timestamp)
context_today already returns strftime value fit for the date field